core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -3713,6 +3714,152 @@ protected PaginatedSearchResult doListUsers(String filter, int limit, int offset
         return result;
     }
 
+    /**
+     * Count users with claim.
+     *
+     * @param claimUri claim uri.
+     * @param value The filter for the user name. Use '*' to have all.
+     * @return Count of the users.
+     * @throws UserStoreException UserStoreException
+     */
+    public long doCountUsersWithClaims(String claimUri, String value) throws UserStoreException {
+
+        if (claimUri == null) {
+            throw new IllegalArgumentException("Error while getting the claim uri");
+        }
+
+        String valueFilter = value;
+        if (valueFilter == null) {
+            throw new IllegalArgumentException("Error while getting the claim filter");
+        }
+
+        String sqlStmt;
+        if (isUserNameClaim(claimUri)) {
+            sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.COUNT_USERS);
+
+        } else {
+            sqlStmt = JDBCRealmConstants.COUNT_USERS_WITH_CLAIM_SQL;
+        }
+
+        if (valueFilter.equals("*")) {
+            valueFilter = "%";
+        } else {
+            valueFilter = valueFilter.trim();
+            valueFilter = valueFilter.replace("*", "%");
+            valueFilter = valueFilter.replace("?", "_");
+        }
+
+        try (Connection dbConnection = getDBConnection();
+             PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
+            String domainName = getMyDomainName();
+            if (StringUtils.isEmpty(domainName)) {
+                domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+            }
+
+            if (isUserNameClaim(claimUri)) {
+                prepStmt.setString(1, valueFilter);
+                prepStmt.setInt(2, tenantId);
+            } else {
+                prepStmt.setString(1, userRealm.getClaimManager().getAttributeName(domainName, claimUri));
+                prepStmt.setInt(2, tenantId);
+                prepStmt.setString(3, valueFilter);
+                prepStmt.setString(4, UserCoreConstants.DEFAULT_PROFILE);
+            }
+
+            ResultSet resultSet = prepStmt.executeQuery();
+            if (resultSet.next()) {
+                return resultSet.getLong("RESULT");
+            } else {
+                log.warn("No result for the filter" + value);
+                return 0;
+            }
+
+        } catch (SQLException e) {
+            String msg = "Error while executing the SQL " + sqlStmt;
+            if (log.isDebugEnabled()) {
+                log.debug(msg + sqlStmt);
+            }
+            throw new UserStoreException(msg, e);
```

Comment:
use error codes

Reply:
fix - in https://github.com/wso2/carbon-kernel/pull/2375